syntax = "proto2";

package rv.protos;

import "google/protobuf/struct.proto";

message RasterTransformerConfig {
    message ChannelTransformerConfig {
        // Configuration for ChannelOrderTransformer
        // The channel indices and order to use when extracting chips from a
        // RasterSource. For instance, this should be an array of length 3 to
        // generate RGB chips.
        repeated int32 channel_order = 2;
    }

    required string transformer_type = 1;

    oneof raster_transformer_config {
        ChannelTransformerConfig channel_transformer_config = 3;

        // Configuration for StatsTransformer
        // File with stats generated by compute_stats command. Used to
        // convert non-uint8 raster values to uint8.
        string stats_uri = 4;

        // Configuration for custom transformers
        google.protobuf.Struct custom_config = 5;
    }
}
